File: 2966.divide-array-into-arrays-with-max-difference/solution.go
func divideArray(nums []int, k int) [][]int {
    var res [][]int
    
    sort.Ints(nums)
    for i := 0; i < len(nums); i += 3 {
        if nums[i+2] - nums[i] > k {
            return nil
        }
        res = append(res, nums[i:i+3])
    }
    return res
}

File: 2966.divide-array-into-arrays-with-max-difference/solution.kt
class Solution {
    fun divideArray(nums: IntArray, k: Int): Array<IntArray> {
        val res = nums
            .sorted()
            .chunked(3)
            .map { it.toIntArray() }
            .toTypedArray()
        return if (res.all { it[2] - it[0] <= k }) {
            res
        } else {
            emptyArray()
        }
    }
}